command/aarelayer/service/aa_relayer_service.go

```diff
@@ -192,8 +197,10 @@ func (rs *AARelayerService) getFirstValidTx() *AAStateTransaction {
 			rs.logger.Debug("transaction nonce mismatch",
 				"tx", poppedTx.ID, "from", address,
 				"nonce", poppedTx.Tx.Transaction.Nonce, "expected", nonce)
-
-			pushBackList = append(pushBackList, poppedTx)
+			// if less then it should not be processed at all
+			if poppedTx.Tx.Transaction.Nonce > nonce {
+				pushBackList = append(pushBackList, poppedTx)
```

Comment:
Just to clarify to myself: `pushBackList` is used to reinject currently invalid transactions back to the pool (since they eventually can become valid ones at some point in the future)? If that's correct, then do we have a mechanism to evict transaction from the pool at some point, if it still is invalid at some later point as well?

Reply:
yes, if the nonce too high than tx is returned to pool for later processing. otherwise, if nonce is too low, status is changed to failed